{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T17:57:50.498878Z",
     "iopub.status.busy": "2023-11-08T17:57:50.498748Z",
     "iopub.status.idle": "2023-11-08T17:57:50.586993Z",
     "shell.execute_reply": "2023-11-08T17:57:50.586380Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-08T17:57:50.589629Z",
     "iopub.status.busy": "2023-11-08T17:57:50.589441Z",
     "iopub.status.idle": "2023-11-08T19:46:47.127579Z",
     "shell.execute_reply": "2023-11-08T19:46:47.126915Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:57:50.778295: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "34937\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:59:17.081704: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.093005: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.123256: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.126657: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.126963: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.127138: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.206728: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.206976: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.207149: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:17.207293: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 12:59:17.207711: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:59:18.121178: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:18.121416: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:18.121586: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:18.121796: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:18.121967: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 12:59:18.122103: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 12:59:18.122135: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 12:59:18.143223: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 12:59:18.356195: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_20/lstm_cell_20/recurrent_kernel/Assign' id:3336 op device:{requested: '', assigned: ''} def:{{{node lstm_20/lstm_cell_20/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_20/lstm_cell_20/recurrent_kernel, lstm_20/lstm_cell_20/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:59:18.486072: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:59:18.518244: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31435, 95)\n",
      "Train on 31435 samples, validate on 3502 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:59:22.856594: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/conv2d_1/bias/v/Assign' id:17012 op device:{requested: '', assigned: ''} def:{{{node training/Adam/conv2d_1/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/conv2d_1/bias/v, training/Adam/conv2d_1/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:59:26.629757: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 12:59:28.720706: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n",
      "2023-11-08 12:59:28.932530: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31435/31435 [==============================] - ETA: 0s - loss: 3.0016"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 12:59:52.732753: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.79418, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 32s 1ms/sample - loss: 3.0016 - val_loss: 1.7942\n",
      "Epoch 2/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.7657\n",
      "Epoch 2: val_loss improved from 1.79418 to 1.59902, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 24s 760us/sample - loss: 1.7657 - val_loss: 1.5990\n",
      "Epoch 3/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.6087\n",
      "Epoch 3: val_loss improved from 1.59902 to 1.53611, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 23s 716us/sample - loss: 1.6087 - val_loss: 1.5361\n",
      "Epoch 4/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.5564\n",
      "Epoch 4: val_loss improved from 1.53611 to 1.51031, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 19s 618us/sample - loss: 1.5564 - val_loss: 1.5103\n",
      "Epoch 5/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.5342\n",
      "Epoch 5: val_loss improved from 1.51031 to 1.49158, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 626us/sample - loss: 1.5342 - val_loss: 1.4916\n",
      "Epoch 6/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.5161\n",
      "Epoch 6: val_loss improved from 1.49158 to 1.47894, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 626us/sample - loss: 1.5161 - val_loss: 1.4789\n",
      "Epoch 7/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.5068\n",
      "Epoch 7: val_loss improved from 1.47894 to 1.47101, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 638us/sample - loss: 1.5068 - val_loss: 1.4710\n",
      "Epoch 8/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.5049\n",
      "Epoch 8: val_loss improved from 1.47101 to 1.46550, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 623us/sample - loss: 1.5049 - val_loss: 1.4655\n",
      "Epoch 9/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4969\n",
      "Epoch 9: val_loss improved from 1.46550 to 1.45823, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 19s 620us/sample - loss: 1.4969 - val_loss: 1.4582\n",
      "Epoch 10/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4966\n",
      "Epoch 10: val_loss did not improve from 1.45823\n",
      "31435/31435 [==============================] - 20s 639us/sample - loss: 1.4966 - val_loss: 1.4606\n",
      "Epoch 11/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.5027\n",
      "Epoch 11: val_loss did not improve from 1.45823\n",
      "31435/31435 [==============================] - 21s 663us/sample - loss: 1.5027 - val_loss: 1.4602\n",
      "Epoch 12/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.5058\n",
      "Epoch 12: val_loss did not improve from 1.45823\n",
      "31435/31435 [==============================] - 20s 651us/sample - loss: 1.5058 - val_loss: 1.4615\n",
      "Epoch 13/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4983\n",
      "Epoch 13: val_loss improved from 1.45823 to 1.44988, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 19s 620us/sample - loss: 1.4983 - val_loss: 1.4499\n",
      "Epoch 14/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4823\n",
      "Epoch 14: val_loss did not improve from 1.44988\n",
      "31435/31435 [==============================] - 21s 676us/sample - loss: 1.4823 - val_loss: 1.4535\n",
      "Epoch 15/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4806\n",
      "Epoch 15: val_loss improved from 1.44988 to 1.44901, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 23s 718us/sample - loss: 1.4806 - val_loss: 1.4490\n",
      "Epoch 16/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4767\n",
      "Epoch 16: val_loss improved from 1.44901 to 1.44154, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 688us/sample - loss: 1.4767 - val_loss: 1.4415\n",
      "Epoch 17/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4818\n",
      "Epoch 17: val_loss did not improve from 1.44154\n",
      "31435/31435 [==============================] - 22s 700us/sample - loss: 1.4818 - val_loss: 1.4445\n",
      "Epoch 18/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4769\n",
      "Epoch 18: val_loss improved from 1.44154 to 1.44100, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 694us/sample - loss: 1.4769 - val_loss: 1.4410\n",
      "Epoch 19/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4726\n",
      "Epoch 19: val_loss did not improve from 1.44100\n",
      "31435/31435 [==============================] - 20s 626us/sample - loss: 1.4726 - val_loss: 1.4440\n",
      "Epoch 20/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4757\n",
      "Epoch 20: val_loss did not improve from 1.44100\n",
      "31435/31435 [==============================] - 21s 661us/sample - loss: 1.4757 - val_loss: 1.4427\n",
      "Epoch 21/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4678\n",
      "Epoch 21: val_loss improved from 1.44100 to 1.43820, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 715us/sample - loss: 1.4678 - val_loss: 1.4382\n",
      "Epoch 22/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4685\n",
      "Epoch 22: val_loss improved from 1.43820 to 1.43689, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 21s 682us/sample - loss: 1.4685 - val_loss: 1.4369\n",
      "Epoch 23/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4621\n",
      "Epoch 23: val_loss improved from 1.43689 to 1.43597, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 23s 718us/sample - loss: 1.4621 - val_loss: 1.4360\n",
      "Epoch 24/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4603\n",
      "Epoch 24: val_loss did not improve from 1.43597\n",
      "31435/31435 [==============================] - 23s 719us/sample - loss: 1.4603 - val_loss: 1.4371\n",
      "Epoch 25/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4581\n",
      "Epoch 25: val_loss improved from 1.43597 to 1.42650, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 21s 677us/sample - loss: 1.4581 - val_loss: 1.4265\n",
      "Epoch 26/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4529\n",
      "Epoch 26: val_loss did not improve from 1.42650\n",
      "31435/31435 [==============================] - 20s 636us/sample - loss: 1.4529 - val_loss: 1.4289\n",
      "Epoch 27/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4490\n",
      "Epoch 27: val_loss did not improve from 1.42650\n",
      "31435/31435 [==============================] - 20s 632us/sample - loss: 1.4490 - val_loss: 1.4283\n",
      "Epoch 28/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4436\n",
      "Epoch 28: val_loss improved from 1.42650 to 1.42419, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 628us/sample - loss: 1.4436 - val_loss: 1.4242\n",
      "Epoch 29/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4426\n",
      "Epoch 29: val_loss did not improve from 1.42419\n",
      "31435/31435 [==============================] - 20s 625us/sample - loss: 1.4426 - val_loss: 1.4285\n",
      "Epoch 30/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4419\n",
      "Epoch 30: val_loss did not improve from 1.42419\n",
      "31435/31435 [==============================] - 19s 619us/sample - loss: 1.4419 - val_loss: 1.4251\n",
      "Epoch 31/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4383\n",
      "Epoch 31: val_loss did not improve from 1.42419\n",
      "31435/31435 [==============================] - 20s 624us/sample - loss: 1.4383 - val_loss: 1.4252\n",
      "Epoch 32/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4373\n",
      "Epoch 32: val_loss did not improve from 1.42419\n",
      "31435/31435 [==============================] - 20s 623us/sample - loss: 1.4373 - val_loss: 1.4322\n",
      "Epoch 33/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4348\n",
      "Epoch 33: val_loss improved from 1.42419 to 1.42093, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 630us/sample - loss: 1.4348 - val_loss: 1.4209\n",
      "Epoch 34/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4322\n",
      "Epoch 34: val_loss improved from 1.42093 to 1.41553, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 626us/sample - loss: 1.4322 - val_loss: 1.4155\n",
      "Epoch 35/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4302\n",
      "Epoch 35: val_loss improved from 1.41553 to 1.41440, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 21s 672us/sample - loss: 1.4302 - val_loss: 1.4144\n",
      "Epoch 36/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4304\n",
      "Epoch 36: val_loss did not improve from 1.41440\n",
      "31435/31435 [==============================] - 19s 617us/sample - loss: 1.4304 - val_loss: 1.4175\n",
      "Epoch 37/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4261\n",
      "Epoch 37: val_loss improved from 1.41440 to 1.41282, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 19s 619us/sample - loss: 1.4261 - val_loss: 1.4128\n",
      "Epoch 38/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4272\n",
      "Epoch 38: val_loss did not improve from 1.41282\n",
      "31435/31435 [==============================] - 23s 718us/sample - loss: 1.4272 - val_loss: 1.4154\n",
      "Epoch 39/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4264\n",
      "Epoch 39: val_loss improved from 1.41282 to 1.41231, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 21s 676us/sample - loss: 1.4264 - val_loss: 1.4123\n",
      "Epoch 40/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4231\n",
      "Epoch 40: val_loss improved from 1.41231 to 1.41061, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 633us/sample - loss: 1.4231 - val_loss: 1.4106\n",
      "Epoch 41/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4217\n",
      "Epoch 41: val_loss did not improve from 1.41061\n",
      "31435/31435 [==============================] - 22s 712us/sample - loss: 1.4217 - val_loss: 1.4147\n",
      "Epoch 42/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4200\n",
      "Epoch 42: val_loss improved from 1.41061 to 1.40948, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 640us/sample - loss: 1.4200 - val_loss: 1.4095\n",
      "Epoch 43/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4177\n",
      "Epoch 43: val_loss did not improve from 1.40948\n",
      "31435/31435 [==============================] - 21s 675us/sample - loss: 1.4177 - val_loss: 1.4098\n",
      "Epoch 44/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4135\n",
      "Epoch 44: val_loss did not improve from 1.40948\n",
      "31435/31435 [==============================] - 22s 685us/sample - loss: 1.4135 - val_loss: 1.4103\n",
      "Epoch 45/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4129\n",
      "Epoch 45: val_loss improved from 1.40948 to 1.40754, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 644us/sample - loss: 1.4129 - val_loss: 1.4075\n",
      "Epoch 46/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4129\n",
      "Epoch 46: val_loss improved from 1.40754 to 1.40718, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 23s 716us/sample - loss: 1.4129 - val_loss: 1.4072\n",
      "Epoch 47/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4104\n",
      "Epoch 47: val_loss improved from 1.40718 to 1.40602, saving model to ./checkpoints/unknown_person_baseline_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 704us/sample - loss: 1.4104 - val_loss: 1.4060\n",
      "Epoch 48/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4096\n",
      "Epoch 48: val_loss did not improve from 1.40602\n",
      "31435/31435 [==============================] - 19s 596us/sample - loss: 1.4096 - val_loss: 1.4081\n",
      "Epoch 49/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4112\n",
      "Epoch 49: val_loss did not improve from 1.40602\n",
      "31435/31435 [==============================] - 20s 628us/sample - loss: 1.4112 - val_loss: 1.4102\n",
      "Epoch 50/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4089\n",
      "Epoch 50: val_loss did not improve from 1.40602\n",
      "31435/31435 [==============================] - 20s 639us/sample - loss: 1.4089 - val_loss: 1.4143\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 13:16:54.963980: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_1_1/kernel/Assign' id:24601 op device:{requested: '', assigned: ''} def:{{{node dense_1_1/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_1_1/kernel, dense_1_1/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 13:16:56.647888: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_29_1/lstm_cell_66/kernel/m/Assign' id:25687 op device:{requested: '', assigned: ''} def:{{{node lstm_29_1/lstm_cell_66/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_29_1/lstm_cell_66/kernel/m, lstm_29_1/lstm_cell_66/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 13:16:57.967753: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 2064)\n",
      "(1514, 2064)\n",
      "(1644, 2064)\n",
      "(1764, 2064)\n",
      "(1836, 2064)\n",
      "(1699, 2064)\n",
      "(1369, 2064)\n",
      "(1766, 2064)\n",
      "(1631, 2064)\n",
      "(1692, 2064)\n",
      "(1550, 2064)\n",
      "(1920, 2064)\n",
      "(1739, 2064)\n",
      "(1788, 2064)\n",
      "(1692, 2064)\n",
      "(1812, 2064)\n",
      "(970, 2064)\n",
      "(1668, 2064)\n",
      "(1896, 2064)\n",
      "{1: 7.180251730260805, 2: 2.5185295250790967, 4: 4.73172376536167, 5: 10.0, 6: 6.504274786961141, 8: 6.0507705909261205, 9: 1.7806968591560808, 10: 7.333192669489142, 11: 3.6387374641662746, 12: 6.37983547000156, 13: 7.195481633922346, 17: 5.85659696093925, 19: 5.382448718370536, 21: 8.073195577579682, 25: 5.918345699793781, 26: 7.640541753918227, 27: 1.0, 28: 7.308192855224852, 29: 5.096332091203663}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2792232/3783125632.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31435 samples, validate on 3502 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 13:20:36.928301: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31435/31435 [==============================] - ETA: 0s - loss: 8.6969\n",
      "Epoch 1: val_loss improved from inf to 1.41575, saving model to ./checkpoints/unknown_person_p22_20.h5\n",
      "31435/31435 [==============================] - 25s 786us/sample - loss: 8.6969 - val_loss: 1.4157\n",
      "Epoch 2/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.6595\n",
      "Epoch 2: val_loss improved from 1.41575 to 1.40128, saving model to ./checkpoints/unknown_person_p22_20.h5\n",
      "31435/31435 [==============================] - 21s 657us/sample - loss: 8.6595 - val_loss: 1.4013\n",
      "Epoch 3/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5758\n",
      "Epoch 3: val_loss did not improve from 1.40128\n",
      "31435/31435 [==============================] - 21s 666us/sample - loss: 8.5758 - val_loss: 1.4041\n",
      "Epoch 4/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5369\n",
      "Epoch 4: val_loss did not improve from 1.40128\n",
      "31435/31435 [==============================] - 22s 713us/sample - loss: 8.5369 - val_loss: 1.4027\n",
      "Epoch 5/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4967\n",
      "Epoch 5: val_loss improved from 1.40128 to 1.39869, saving model to ./checkpoints/unknown_person_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 687us/sample - loss: 8.4967 - val_loss: 1.3987\n",
      "Epoch 6/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4814\n",
      "Epoch 6: val_loss did not improve from 1.39869\n",
      "31435/31435 [==============================] - 22s 711us/sample - loss: 8.4814 - val_loss: 1.4032\n",
      "Epoch 7/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4699\n",
      "Epoch 7: val_loss did not improve from 1.39869\n",
      "31435/31435 [==============================] - 22s 698us/sample - loss: 8.4699 - val_loss: 1.3992\n",
      "Epoch 8/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4390\n",
      "Epoch 8: val_loss did not improve from 1.39869\n",
      "31435/31435 [==============================] - 19s 615us/sample - loss: 8.4390 - val_loss: 1.4021\n",
      "Epoch 9/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4360\n",
      "Epoch 9: val_loss did not improve from 1.39869\n",
      "31435/31435 [==============================] - 19s 617us/sample - loss: 8.4360 - val_loss: 1.4002\n",
      "Epoch 10/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4330\n",
      "Epoch 10: val_loss improved from 1.39869 to 1.39574, saving model to ./checkpoints/unknown_person_p22_20.h5\n",
      "31435/31435 [==============================] - 19s 618us/sample - loss: 8.4330 - val_loss: 1.3957\n",
      "Epoch 11/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4131\n",
      "Epoch 11: val_loss did not improve from 1.39574\n",
      "31435/31435 [==============================] - 21s 653us/sample - loss: 8.4131 - val_loss: 1.3966\n",
      "Epoch 12/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4002\n",
      "Epoch 12: val_loss did not improve from 1.39574\n",
      "31435/31435 [==============================] - 21s 673us/sample - loss: 8.4002 - val_loss: 1.3979\n",
      "Epoch 13/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3830\n",
      "Epoch 13: val_loss improved from 1.39574 to 1.38959, saving model to ./checkpoints/unknown_person_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 624us/sample - loss: 8.3830 - val_loss: 1.3896\n",
      "Epoch 14/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3668\n",
      "Epoch 14: val_loss did not improve from 1.38959\n",
      "31435/31435 [==============================] - 20s 635us/sample - loss: 8.3668 - val_loss: 1.3930\n",
      "Epoch 15/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3504\n",
      "Epoch 15: val_loss did not improve from 1.38959\n",
      "31435/31435 [==============================] - 19s 618us/sample - loss: 8.3504 - val_loss: 1.3917\n",
      "Epoch 16/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3358\n",
      "Epoch 16: val_loss did not improve from 1.38959\n",
      "31435/31435 [==============================] - 19s 613us/sample - loss: 8.3358 - val_loss: 1.3944\n",
      "Epoch 17/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3262\n",
      "Epoch 17: val_loss did not improve from 1.38959\n",
      "31435/31435 [==============================] - 19s 619us/sample - loss: 8.3262 - val_loss: 1.3904\n",
      "Epoch 18/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3335\n",
      "Epoch 18: val_loss improved from 1.38959 to 1.38546, saving model to ./checkpoints/unknown_person_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 634us/sample - loss: 8.3335 - val_loss: 1.3855\n",
      "Epoch 19/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3296\n",
      "Epoch 19: val_loss did not improve from 1.38546\n",
      "31435/31435 [==============================] - 19s 611us/sample - loss: 8.3296 - val_loss: 1.3885\n",
      "Epoch 20/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3056\n",
      "Epoch 20: val_loss did not improve from 1.38546\n",
      "31435/31435 [==============================] - 20s 647us/sample - loss: 8.3056 - val_loss: 1.3981\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 13:27:30.357658: W tensorflow/c/c_api.cc:304] Operation '{name:'conv2d_2_2/bias/Assign' id:37819 op device:{requested: '', assigned: ''} def:{{{node conv2d_2_2/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](conv2d_2_2/bias, conv2d_2_2/bias/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 13:27:32.806038: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_3_2/bias/v/Assign' id:45889 op device:{requested: '', assigned: ''} def:{{{node dense_3_2/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_3_2/bias/v, dense_3_2/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31435 samples, validate on 3502 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 13:27:37.749105: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 13:27:50.716445: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4097"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 13:28:11.675275: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.41001, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 27s 869us/sample - loss: 1.4097 - val_loss: 1.4100\n",
      "Epoch 2/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4098\n",
      "Epoch 2: val_loss improved from 1.41001 to 1.40724, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 705us/sample - loss: 1.4098 - val_loss: 1.4072\n",
      "Epoch 3/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4071\n",
      "Epoch 3: val_loss improved from 1.40724 to 1.40715, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 23s 717us/sample - loss: 1.4071 - val_loss: 1.4071\n",
      "Epoch 4/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4063\n",
      "Epoch 4: val_loss improved from 1.40715 to 1.40305, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 20s 651us/sample - loss: 1.4063 - val_loss: 1.4031\n",
      "Epoch 5/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4017\n",
      "Epoch 5: val_loss did not improve from 1.40305\n",
      "31435/31435 [==============================] - 19s 606us/sample - loss: 1.4017 - val_loss: 1.4043\n",
      "Epoch 6/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4024\n",
      "Epoch 6: val_loss improved from 1.40305 to 1.40040, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.4024 - val_loss: 1.4004\n",
      "Epoch 7/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3999\n",
      "Epoch 7: val_loss did not improve from 1.40040\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.3999 - val_loss: 1.4071\n",
      "Epoch 8/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3996\n",
      "Epoch 8: val_loss did not improve from 1.40040\n",
      "31435/31435 [==============================] - 20s 640us/sample - loss: 1.3996 - val_loss: 1.4030\n",
      "Epoch 9/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3963\n",
      "Epoch 9: val_loss did not improve from 1.40040\n",
      "31435/31435 [==============================] - 22s 706us/sample - loss: 1.3963 - val_loss: 1.4009\n",
      "Epoch 10/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3996\n",
      "Epoch 10: val_loss did not improve from 1.40040\n",
      "31435/31435 [==============================] - 22s 707us/sample - loss: 1.3996 - val_loss: 1.4023\n",
      "Epoch 11/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3954\n",
      "Epoch 11: val_loss did not improve from 1.40040\n",
      "31435/31435 [==============================] - 22s 703us/sample - loss: 1.3954 - val_loss: 1.4011\n",
      "Epoch 12/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3918\n",
      "Epoch 12: val_loss improved from 1.40040 to 1.39854, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 708us/sample - loss: 1.3918 - val_loss: 1.3985\n",
      "Epoch 13/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3904\n",
      "Epoch 13: val_loss improved from 1.39854 to 1.39835, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 713us/sample - loss: 1.3904 - val_loss: 1.3983\n",
      "Epoch 14/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3923\n",
      "Epoch 14: val_loss improved from 1.39835 to 1.39704, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 707us/sample - loss: 1.3923 - val_loss: 1.3970\n",
      "Epoch 15/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3888\n",
      "Epoch 15: val_loss improved from 1.39704 to 1.39533, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 22s 696us/sample - loss: 1.3888 - val_loss: 1.3953\n",
      "Epoch 16/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3890\n",
      "Epoch 16: val_loss did not improve from 1.39533\n",
      "31435/31435 [==============================] - 20s 628us/sample - loss: 1.3890 - val_loss: 1.3971\n",
      "Epoch 17/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3876\n",
      "Epoch 17: val_loss did not improve from 1.39533\n",
      "31435/31435 [==============================] - 22s 705us/sample - loss: 1.3876 - val_loss: 1.3977\n",
      "Epoch 18/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3870\n",
      "Epoch 18: val_loss did not improve from 1.39533\n",
      "31435/31435 [==============================] - 21s 662us/sample - loss: 1.3870 - val_loss: 1.3997\n",
      "Epoch 19/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3853\n",
      "Epoch 19: val_loss improved from 1.39533 to 1.39470, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 19s 603us/sample - loss: 1.3853 - val_loss: 1.3947\n",
      "Epoch 20/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3849\n",
      "Epoch 20: val_loss improved from 1.39470 to 1.39114, saving model to ./checkpoints/unknown_person_baseline_2_p22_20.h5\n",
      "31435/31435 [==============================] - 19s 611us/sample - loss: 1.3849 - val_loss: 1.3911\n",
      "Train on 31435 samples, validate on 3502 samples\n",
      "Epoch 1/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3836\n",
      "Epoch 1: val_loss improved from inf to 1.39283, saving model to ./checkpoints/unknown_person_baseline_p22_21.h5\n",
      "31435/31435 [==============================] - 22s 695us/sample - loss: 1.3836 - val_loss: 1.3928\n",
      "Epoch 2/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3831\n",
      "Epoch 2: val_loss did not improve from 1.39283\n",
      "31435/31435 [==============================] - 20s 633us/sample - loss: 1.3831 - val_loss: 1.3992\n",
      "Epoch 3/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3795\n",
      "Epoch 3: val_loss improved from 1.39283 to 1.39102, saving model to ./checkpoints/unknown_person_baseline_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 608us/sample - loss: 1.3795 - val_loss: 1.3910\n",
      "Epoch 4/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3786\n",
      "Epoch 4: val_loss did not improve from 1.39102\n",
      "31435/31435 [==============================] - 19s 599us/sample - loss: 1.3786 - val_loss: 1.3939\n",
      "Epoch 5/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3784\n",
      "Epoch 5: val_loss improved from 1.39102 to 1.38970, saving model to ./checkpoints/unknown_person_baseline_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 615us/sample - loss: 1.3784 - val_loss: 1.3897\n",
      "Epoch 6/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3762\n",
      "Epoch 6: val_loss did not improve from 1.38970\n",
      "31435/31435 [==============================] - 19s 597us/sample - loss: 1.3762 - val_loss: 1.3910\n",
      "Epoch 7/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3728\n",
      "Epoch 7: val_loss improved from 1.38970 to 1.38918, saving model to ./checkpoints/unknown_person_baseline_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 611us/sample - loss: 1.3728 - val_loss: 1.3892\n",
      "Epoch 8/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3786\n",
      "Epoch 8: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.3786 - val_loss: 1.3916\n",
      "Epoch 9/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3831\n",
      "Epoch 9: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 612us/sample - loss: 1.3831 - val_loss: 1.3916\n",
      "Epoch 10/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3865\n",
      "Epoch 10: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 604us/sample - loss: 1.3865 - val_loss: 1.3900\n",
      "Epoch 11/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3853\n",
      "Epoch 11: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 603us/sample - loss: 1.3853 - val_loss: 1.3964\n",
      "Epoch 12/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3857\n",
      "Epoch 12: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.3857 - val_loss: 1.3908\n",
      "Epoch 13/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3941\n",
      "Epoch 13: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 605us/sample - loss: 1.3941 - val_loss: 1.3905\n",
      "Epoch 14/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3905\n",
      "Epoch 14: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 613us/sample - loss: 1.3905 - val_loss: 1.3957\n",
      "Epoch 15/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3991\n",
      "Epoch 15: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 694us/sample - loss: 1.3991 - val_loss: 1.3921\n",
      "Epoch 16/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3983\n",
      "Epoch 16: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 711us/sample - loss: 1.3983 - val_loss: 1.3968\n",
      "Epoch 17/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4303\n",
      "Epoch 17: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 703us/sample - loss: 1.4303 - val_loss: 1.4088\n",
      "Epoch 18/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4021\n",
      "Epoch 18: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 705us/sample - loss: 1.4021 - val_loss: 1.3997\n",
      "Epoch 19/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4062\n",
      "Epoch 19: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 703us/sample - loss: 1.4062 - val_loss: 1.4014\n",
      "Epoch 20/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4089\n",
      "Epoch 20: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 706us/sample - loss: 1.4089 - val_loss: 1.4015\n",
      "Epoch 21/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4158\n",
      "Epoch 21: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 712us/sample - loss: 1.4158 - val_loss: 1.4162\n",
      "Epoch 22/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4124\n",
      "Epoch 22: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 20s 631us/sample - loss: 1.4124 - val_loss: 1.4170\n",
      "Epoch 23/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4092\n",
      "Epoch 23: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 693us/sample - loss: 1.4092 - val_loss: 1.4087\n",
      "Epoch 24/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4041\n",
      "Epoch 24: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 695us/sample - loss: 1.4041 - val_loss: 1.4133\n",
      "Epoch 25/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4665\n",
      "Epoch 25: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 21s 657us/sample - loss: 1.4665 - val_loss: 1.4086\n",
      "Epoch 26/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4205\n",
      "Epoch 26: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 21s 671us/sample - loss: 1.4205 - val_loss: 1.4131\n",
      "Epoch 27/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4067\n",
      "Epoch 27: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 704us/sample - loss: 1.4067 - val_loss: 1.4059\n",
      "Epoch 28/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4039\n",
      "Epoch 28: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 710us/sample - loss: 1.4039 - val_loss: 1.4088\n",
      "Epoch 29/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4034\n",
      "Epoch 29: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 707us/sample - loss: 1.4034 - val_loss: 1.4139\n",
      "Epoch 30/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4016\n",
      "Epoch 30: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 20s 642us/sample - loss: 1.4016 - val_loss: 1.4039\n",
      "Epoch 31/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3983\n",
      "Epoch 31: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 20s 630us/sample - loss: 1.3983 - val_loss: 1.4055\n",
      "Epoch 32/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3955\n",
      "Epoch 32: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 21s 666us/sample - loss: 1.3955 - val_loss: 1.4038\n",
      "Epoch 33/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3913\n",
      "Epoch 33: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 704us/sample - loss: 1.3913 - val_loss: 1.4022\n",
      "Epoch 34/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3943\n",
      "Epoch 34: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 702us/sample - loss: 1.3943 - val_loss: 1.4045\n",
      "Epoch 35/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3903\n",
      "Epoch 35: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 695us/sample - loss: 1.3903 - val_loss: 1.4000\n",
      "Epoch 36/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3909\n",
      "Epoch 36: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 21s 652us/sample - loss: 1.3909 - val_loss: 1.4058\n",
      "Epoch 37/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3891\n",
      "Epoch 37: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 22s 690us/sample - loss: 1.3891 - val_loss: 1.4006\n",
      "Epoch 38/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3891\n",
      "Epoch 38: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 613us/sample - loss: 1.3891 - val_loss: 1.4034\n",
      "Epoch 39/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3853\n",
      "Epoch 39: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.3853 - val_loss: 1.4018\n",
      "Epoch 40/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3850\n",
      "Epoch 40: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 610us/sample - loss: 1.3850 - val_loss: 1.3995\n",
      "Epoch 41/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3828\n",
      "Epoch 41: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 606us/sample - loss: 1.3828 - val_loss: 1.4032\n",
      "Epoch 42/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3795\n",
      "Epoch 42: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 20s 629us/sample - loss: 1.3795 - val_loss: 1.4014\n",
      "Epoch 43/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3838\n",
      "Epoch 43: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 21s 662us/sample - loss: 1.3838 - val_loss: 1.3948\n",
      "Epoch 44/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3803\n",
      "Epoch 44: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 21s 683us/sample - loss: 1.3803 - val_loss: 1.3970\n",
      "Epoch 45/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3791\n",
      "Epoch 45: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 604us/sample - loss: 1.3791 - val_loss: 1.4026\n",
      "Epoch 46/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3780\n",
      "Epoch 46: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 20s 621us/sample - loss: 1.3780 - val_loss: 1.3962\n",
      "Epoch 47/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3778\n",
      "Epoch 47: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 607us/sample - loss: 1.3778 - val_loss: 1.3925\n",
      "Epoch 48/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3750\n",
      "Epoch 48: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 604us/sample - loss: 1.3750 - val_loss: 1.3941\n",
      "Epoch 49/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3751\n",
      "Epoch 49: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.3751 - val_loss: 1.3960\n",
      "Epoch 50/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3750\n",
      "Epoch 50: val_loss did not improve from 1.38918\n",
      "31435/31435 [==============================] - 20s 627us/sample - loss: 1.3750 - val_loss: 1.3949\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 13:51:59.612610: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_3_3/bias/Assign' id:63406 op device:{requested: '', assigned: ''} def:{{{node dense_3_3/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_3_3/bias, dense_3_3/bias/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 13:52:02.855552: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_31_3/lstm_cell_142/recurrent_kernel/m/Assign' id:64480 op device:{requested: '', assigned: ''} def:{{{node lstm_31_3/lstm_cell_142/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_31_3/lstm_cell_142/recurrent_kernel/m, lstm_31_3/lstm_cell_142/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 13:52:05.942369: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 2064)\n",
      "(1514, 2064)\n",
      "(1644, 2064)\n",
      "(1764, 2064)\n",
      "(1836, 2064)\n",
      "(1699, 2064)\n",
      "(1369, 2064)\n",
      "(1766, 2064)\n",
      "(1631, 2064)\n",
      "(1692, 2064)\n",
      "(1550, 2064)\n",
      "(1920, 2064)\n",
      "(1739, 2064)\n",
      "(1788, 2064)\n",
      "(1692, 2064)\n",
      "(1812, 2064)\n",
      "(970, 2064)\n",
      "(1668, 2064)\n",
      "(1896, 2064)\n",
      "{1: 7.2084190393824334, 2: 2.4546279960102186, 4: 5.5380293171843435, 5: 10.0, 6: 6.592311375383677, 8: 6.464593304248255, 9: 1.9796774600221698, 10: 7.7635394872726105, 11: 4.085923348907938, 12: 6.889926324913728, 13: 7.011595056799513, 17: 6.475179158030741, 19: 5.750332318469199, 21: 8.520312415287368, 25: 6.046387412462253, 26: 7.522690737737204, 27: 1.0, 28: 6.937991823169342, 29: 4.032147525231709}\n",
      "Train on 31435 samples, validate on 3502 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 13:56:07.336363: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31435/31435 [==============================] - ETA: 0s - loss: 8.7275\n",
      "Epoch 1: val_loss improved from inf to 1.41353, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 32s 1ms/sample - loss: 8.7275 - val_loss: 1.4135\n",
      "Epoch 2/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.6759\n",
      "Epoch 2: val_loss improved from 1.41353 to 1.40536, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 22s 703us/sample - loss: 8.6759 - val_loss: 1.4054\n",
      "Epoch 3/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.6789\n",
      "Epoch 3: val_loss improved from 1.40536 to 1.39378, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 20s 636us/sample - loss: 8.6789 - val_loss: 1.3938\n",
      "Epoch 4/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.6185\n",
      "Epoch 4: val_loss did not improve from 1.39378\n",
      "31435/31435 [==============================] - 19s 600us/sample - loss: 8.6185 - val_loss: 1.3942\n",
      "Epoch 5/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5964\n",
      "Epoch 5: val_loss did not improve from 1.39378\n",
      "31435/31435 [==============================] - 21s 674us/sample - loss: 8.5964 - val_loss: 1.4004\n",
      "Epoch 6/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5668\n",
      "Epoch 6: val_loss improved from 1.39378 to 1.38909, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 612us/sample - loss: 8.5668 - val_loss: 1.3891\n",
      "Epoch 7/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5466\n",
      "Epoch 7: val_loss did not improve from 1.38909\n",
      "31435/31435 [==============================] - 19s 611us/sample - loss: 8.5466 - val_loss: 1.3935\n",
      "Epoch 8/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5214\n",
      "Epoch 8: val_loss did not improve from 1.38909\n",
      "31435/31435 [==============================] - 20s 622us/sample - loss: 8.5214 - val_loss: 1.3910\n",
      "Epoch 9/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5115\n",
      "Epoch 9: val_loss improved from 1.38909 to 1.38565, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 22s 710us/sample - loss: 8.5115 - val_loss: 1.3857\n",
      "Epoch 10/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5038\n",
      "Epoch 10: val_loss did not improve from 1.38565\n",
      "31435/31435 [==============================] - 21s 670us/sample - loss: 8.5038 - val_loss: 1.4005\n",
      "Epoch 11/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4862\n",
      "Epoch 11: val_loss did not improve from 1.38565\n",
      "31435/31435 [==============================] - 22s 705us/sample - loss: 8.4862 - val_loss: 1.3893\n",
      "Epoch 12/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4771\n",
      "Epoch 12: val_loss did not improve from 1.38565\n",
      "31435/31435 [==============================] - 22s 696us/sample - loss: 8.4771 - val_loss: 1.3971\n",
      "Epoch 13/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4589\n",
      "Epoch 13: val_loss did not improve from 1.38565\n",
      "31435/31435 [==============================] - 22s 700us/sample - loss: 8.4589 - val_loss: 1.3865\n",
      "Epoch 14/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4608\n",
      "Epoch 14: val_loss improved from 1.38565 to 1.38433, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 20s 638us/sample - loss: 8.4608 - val_loss: 1.3843\n",
      "Epoch 15/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4394\n",
      "Epoch 15: val_loss did not improve from 1.38433\n",
      "31435/31435 [==============================] - 19s 605us/sample - loss: 8.4394 - val_loss: 1.3863\n",
      "Epoch 16/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4371\n",
      "Epoch 16: val_loss did not improve from 1.38433\n",
      "31435/31435 [==============================] - 19s 604us/sample - loss: 8.4371 - val_loss: 1.3858\n",
      "Epoch 17/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4408\n",
      "Epoch 17: val_loss improved from 1.38433 to 1.38327, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 611us/sample - loss: 8.4408 - val_loss: 1.3833\n",
      "Epoch 18/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4141\n",
      "Epoch 18: val_loss improved from 1.38327 to 1.38172, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 610us/sample - loss: 8.4141 - val_loss: 1.3817\n",
      "Epoch 19/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3968\n",
      "Epoch 19: val_loss did not improve from 1.38172\n",
      "31435/31435 [==============================] - 22s 709us/sample - loss: 8.3968 - val_loss: 1.3940\n",
      "Epoch 20/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3930\n",
      "Epoch 20: val_loss improved from 1.38172 to 1.38093, saving model to ./checkpoints/unknown_person_p22_21.h5\n",
      "31435/31435 [==============================] - 22s 707us/sample - loss: 8.3930 - val_loss: 1.3809\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:03:11.707383: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_18_4/lstm_cell_166/bias/Assign' id:79705 op device:{requested: '', assigned: ''} def:{{{node lstm_18_4/lstm_cell_166/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_18_4/lstm_cell_166/bias, lstm_18_4/lstm_cell_166/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 14:03:15.858626: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_25_4/lstm_cell_173/recurrent_kernel/v/Assign' id:84431 op device:{requested: '', assigned: ''} def:{{{node lstm_25_4/lstm_cell_173/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_25_4/lstm_cell_173/recurrent_kernel/v, lstm_25_4/lstm_cell_173/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31435 samples, validate on 3502 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:03:22.455099: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:03:44.296674: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3774"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:04:05.137718: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.39062, saving model to ./checkpoints/unknown_person_baseline_2_p22_21.h5\n",
      "31435/31435 [==============================] - 30s 955us/sample - loss: 1.3774 - val_loss: 1.3906\n",
      "Epoch 2/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3723\n",
      "Epoch 2: val_loss improved from 1.39062 to 1.38719, saving model to ./checkpoints/unknown_person_baseline_2_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 611us/sample - loss: 1.3723 - val_loss: 1.3872\n",
      "Epoch 3/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3798\n",
      "Epoch 3: val_loss improved from 1.38719 to 1.38549, saving model to ./checkpoints/unknown_person_baseline_2_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 613us/sample - loss: 1.3798 - val_loss: 1.3855\n",
      "Epoch 4/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3737\n",
      "Epoch 4: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 19s 613us/sample - loss: 1.3737 - val_loss: 1.3937\n",
      "Epoch 5/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3728\n",
      "Epoch 5: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 20s 621us/sample - loss: 1.3728 - val_loss: 1.3870\n",
      "Epoch 6/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3711\n",
      "Epoch 6: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 19s 608us/sample - loss: 1.3711 - val_loss: 1.3899\n",
      "Epoch 7/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3733\n",
      "Epoch 7: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 20s 628us/sample - loss: 1.3733 - val_loss: 1.3895\n",
      "Epoch 8/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3711\n",
      "Epoch 8: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 21s 674us/sample - loss: 1.3711 - val_loss: 1.3875\n",
      "Epoch 9/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3681\n",
      "Epoch 9: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 20s 625us/sample - loss: 1.3681 - val_loss: 1.3891\n",
      "Epoch 10/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3695\n",
      "Epoch 10: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 20s 633us/sample - loss: 1.3695 - val_loss: 1.3891\n",
      "Epoch 11/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3683\n",
      "Epoch 11: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 19s 605us/sample - loss: 1.3683 - val_loss: 1.3865\n",
      "Epoch 12/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3668\n",
      "Epoch 12: val_loss did not improve from 1.38549\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.3668 - val_loss: 1.3875\n",
      "Epoch 13/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3662\n",
      "Epoch 13: val_loss improved from 1.38549 to 1.38467, saving model to ./checkpoints/unknown_person_baseline_2_p22_21.h5\n",
      "31435/31435 [==============================] - 20s 625us/sample - loss: 1.3662 - val_loss: 1.3847\n",
      "Epoch 14/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3645\n",
      "Epoch 14: val_loss did not improve from 1.38467\n",
      "31435/31435 [==============================] - 19s 614us/sample - loss: 1.3645 - val_loss: 1.3869\n",
      "Epoch 15/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3633\n",
      "Epoch 15: val_loss did not improve from 1.38467\n",
      "31435/31435 [==============================] - 19s 613us/sample - loss: 1.3633 - val_loss: 1.3920\n",
      "Epoch 16/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3603\n",
      "Epoch 16: val_loss improved from 1.38467 to 1.38409, saving model to ./checkpoints/unknown_person_baseline_2_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 615us/sample - loss: 1.3603 - val_loss: 1.3841\n",
      "Epoch 17/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3619\n",
      "Epoch 17: val_loss did not improve from 1.38409\n",
      "31435/31435 [==============================] - 19s 607us/sample - loss: 1.3619 - val_loss: 1.3852\n",
      "Epoch 18/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3639\n",
      "Epoch 18: val_loss did not improve from 1.38409\n",
      "31435/31435 [==============================] - 19s 613us/sample - loss: 1.3639 - val_loss: 1.3846\n",
      "Epoch 19/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3570\n",
      "Epoch 19: val_loss improved from 1.38409 to 1.38097, saving model to ./checkpoints/unknown_person_baseline_2_p22_21.h5\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.3570 - val_loss: 1.3810\n",
      "Epoch 20/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3616\n",
      "Epoch 20: val_loss did not improve from 1.38097\n",
      "31435/31435 [==============================] - 19s 602us/sample - loss: 1.3616 - val_loss: 1.3834\n",
      "Train on 31435 samples, validate on 3502 samples\n",
      "Epoch 1/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3567\n",
      "Epoch 1: val_loss improved from inf to 1.38112, saving model to ./checkpoints/unknown_person_baseline_p22_22.h5\n",
      "31435/31435 [==============================] - 19s 603us/sample - loss: 1.3567 - val_loss: 1.3811\n",
      "Epoch 2/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3537\n",
      "Epoch 2: val_loss did not improve from 1.38112\n",
      "31435/31435 [==============================] - 19s 598us/sample - loss: 1.3537 - val_loss: 1.3818\n",
      "Epoch 3/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3565\n",
      "Epoch 3: val_loss did not improve from 1.38112\n",
      "31435/31435 [==============================] - 19s 612us/sample - loss: 1.3565 - val_loss: 1.3837\n",
      "Epoch 4/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3526\n",
      "Epoch 4: val_loss improved from 1.38112 to 1.37360, saving model to ./checkpoints/unknown_person_baseline_p22_22.h5\n",
      "31435/31435 [==============================] - 19s 603us/sample - loss: 1.3526 - val_loss: 1.3736\n",
      "Epoch 5/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3540\n",
      "Epoch 5: val_loss did not improve from 1.37360\n",
      "31435/31435 [==============================] - 19s 603us/sample - loss: 1.3540 - val_loss: 1.3787\n",
      "Epoch 6/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3497\n",
      "Epoch 6: val_loss did not improve from 1.37360\n",
      "31435/31435 [==============================] - 19s 604us/sample - loss: 1.3497 - val_loss: 1.3802\n",
      "Epoch 7/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3501\n",
      "Epoch 7: val_loss improved from 1.37360 to 1.37293, saving model to ./checkpoints/unknown_person_baseline_p22_22.h5\n",
      "31435/31435 [==============================] - 19s 608us/sample - loss: 1.3501 - val_loss: 1.3729\n",
      "Epoch 8/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3604\n",
      "Epoch 8: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 20s 651us/sample - loss: 1.3604 - val_loss: 1.3815\n",
      "Epoch 9/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3641\n",
      "Epoch 9: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 689us/sample - loss: 1.3641 - val_loss: 1.3843\n",
      "Epoch 10/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3654\n",
      "Epoch 10: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 21s 683us/sample - loss: 1.3654 - val_loss: 1.3819\n",
      "Epoch 11/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3738\n",
      "Epoch 11: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.3738 - val_loss: 1.3858\n",
      "Epoch 12/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3825\n",
      "Epoch 12: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 701us/sample - loss: 1.3825 - val_loss: 1.3919\n",
      "Epoch 13/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3848\n",
      "Epoch 13: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 21s 665us/sample - loss: 1.3848 - val_loss: 1.3960\n",
      "Epoch 14/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3960\n",
      "Epoch 14: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 611us/sample - loss: 1.3960 - val_loss: 1.3903\n",
      "Epoch 15/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3847\n",
      "Epoch 15: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 20s 650us/sample - loss: 1.3847 - val_loss: 1.3938\n",
      "Epoch 16/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4086\n",
      "Epoch 16: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 616us/sample - loss: 1.4086 - val_loss: 1.3916\n",
      "Epoch 17/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4072\n",
      "Epoch 17: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 21s 659us/sample - loss: 1.4072 - val_loss: 1.4002\n",
      "Epoch 18/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3975\n",
      "Epoch 18: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 703us/sample - loss: 1.3975 - val_loss: 1.3993\n",
      "Epoch 19/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4019\n",
      "Epoch 19: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 21s 682us/sample - loss: 1.4019 - val_loss: 1.3964\n",
      "Epoch 20/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4083\n",
      "Epoch 20: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 20s 623us/sample - loss: 1.4083 - val_loss: 1.4049\n",
      "Epoch 21/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4175\n",
      "Epoch 21: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 21s 682us/sample - loss: 1.4175 - val_loss: 1.4064\n",
      "Epoch 22/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4945\n",
      "Epoch 22: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 709us/sample - loss: 1.4945 - val_loss: 1.4124\n",
      "Epoch 23/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4970\n",
      "Epoch 23: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 703us/sample - loss: 1.4970 - val_loss: 1.4278\n",
      "Epoch 24/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4605\n",
      "Epoch 24: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 21s 664us/sample - loss: 1.4605 - val_loss: 1.4078\n",
      "Epoch 25/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4406\n",
      "Epoch 25: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 21s 672us/sample - loss: 1.4406 - val_loss: 1.4132\n",
      "Epoch 26/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4254\n",
      "Epoch 26: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 701us/sample - loss: 1.4254 - val_loss: 1.4081\n",
      "Epoch 27/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4190\n",
      "Epoch 27: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 20s 632us/sample - loss: 1.4190 - val_loss: 1.4028\n",
      "Epoch 28/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4168\n",
      "Epoch 28: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 606us/sample - loss: 1.4168 - val_loss: 1.4053\n",
      "Epoch 29/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4123\n",
      "Epoch 29: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 605us/sample - loss: 1.4123 - val_loss: 1.4058\n",
      "Epoch 30/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4078\n",
      "Epoch 30: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 608us/sample - loss: 1.4078 - val_loss: 1.4045\n",
      "Epoch 31/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4108\n",
      "Epoch 31: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 603us/sample - loss: 1.4108 - val_loss: 1.4049\n",
      "Epoch 32/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4075\n",
      "Epoch 32: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 609us/sample - loss: 1.4075 - val_loss: 1.4013\n",
      "Epoch 33/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4076\n",
      "Epoch 33: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 607us/sample - loss: 1.4076 - val_loss: 1.4040\n",
      "Epoch 34/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4048\n",
      "Epoch 34: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 20s 648us/sample - loss: 1.4048 - val_loss: 1.4046\n",
      "Epoch 35/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4011\n",
      "Epoch 35: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 701us/sample - loss: 1.4011 - val_loss: 1.4027\n",
      "Epoch 36/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3992\n",
      "Epoch 36: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 702us/sample - loss: 1.3992 - val_loss: 1.4021\n",
      "Epoch 37/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.4014\n",
      "Epoch 37: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 690us/sample - loss: 1.4014 - val_loss: 1.4023\n",
      "Epoch 38/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3978\n",
      "Epoch 38: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 617us/sample - loss: 1.3978 - val_loss: 1.4001\n",
      "Epoch 39/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3999\n",
      "Epoch 39: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 20s 621us/sample - loss: 1.3999 - val_loss: 1.3976\n",
      "Epoch 40/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3973\n",
      "Epoch 40: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 20s 622us/sample - loss: 1.3973 - val_loss: 1.3969\n",
      "Epoch 41/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3961\n",
      "Epoch 41: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 600us/sample - loss: 1.3961 - val_loss: 1.4013\n",
      "Epoch 42/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3942\n",
      "Epoch 42: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 602us/sample - loss: 1.3942 - val_loss: 1.4005\n",
      "Epoch 43/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3937\n",
      "Epoch 43: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 21s 683us/sample - loss: 1.3937 - val_loss: 1.3983\n",
      "Epoch 44/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3924\n",
      "Epoch 44: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 705us/sample - loss: 1.3924 - val_loss: 1.4018\n",
      "Epoch 45/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3909\n",
      "Epoch 45: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 702us/sample - loss: 1.3909 - val_loss: 1.4055\n",
      "Epoch 46/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3869\n",
      "Epoch 46: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 22s 688us/sample - loss: 1.3869 - val_loss: 1.3982\n",
      "Epoch 47/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3922\n",
      "Epoch 47: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 603us/sample - loss: 1.3922 - val_loss: 1.3985\n",
      "Epoch 48/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3903\n",
      "Epoch 48: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 598us/sample - loss: 1.3903 - val_loss: 1.3977\n",
      "Epoch 49/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3860\n",
      "Epoch 49: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 601us/sample - loss: 1.3860 - val_loss: 1.3998\n",
      "Epoch 50/50\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3857\n",
      "Epoch 50: val_loss did not improve from 1.37293\n",
      "31435/31435 [==============================] - 19s 600us/sample - loss: 1.3857 - val_loss: 1.3906\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:27:17.119100: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_32_5/lstm_cell_217/bias/Assign' id:101307 op device:{requested: '', assigned: ''} def:{{{node lstm_32_5/lstm_cell_217/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_32_5/lstm_cell_217/bias, lstm_32_5/lstm_cell_217/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 14:27:22.215503: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_24_5/lstm_cell_209/recurrent_kernel/m/Assign' id:103133 op device:{requested: '', assigned: ''} def:{{{node lstm_24_5/lstm_cell_209/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_24_5/lstm_cell_209/recurrent_kernel/m, lstm_24_5/lstm_cell_209/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 14:27:27.168216: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 2064)\n",
      "(1514, 2064)\n",
      "(1644, 2064)\n",
      "(1764, 2064)\n",
      "(1836, 2064)\n",
      "(1699, 2064)\n",
      "(1369, 2064)\n",
      "(1766, 2064)\n",
      "(1631, 2064)\n",
      "(1692, 2064)\n",
      "(1550, 2064)\n",
      "(1920, 2064)\n",
      "(1739, 2064)\n",
      "(1788, 2064)\n",
      "(1692, 2064)\n",
      "(1812, 2064)\n",
      "(970, 2064)\n",
      "(1668, 2064)\n",
      "(1896, 2064)\n",
      "{1: 6.857247851636459, 2: 2.4961956207995617, 4: 5.916998946385706, 5: 10.0, 6: 6.348796176369387, 8: 6.355783905056204, 9: 2.1492810983426747, 10: 7.526463062846325, 11: 3.8920453519537133, 12: 6.925825095958246, 13: 6.843975058103917, 17: 6.643647199173792, 19: 5.934794423142995, 21: 8.4986934724996, 25: 6.252011670882735, 26: 7.566313486746153, 27: 1.0, 28: 6.9697289975863255, 29: 2.824859193846465}\n",
      "Train on 31435 samples, validate on 3502 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:31:34.931586: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5776\n",
      "Epoch 1: val_loss improved from inf to 1.40702, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 31s 995us/sample - loss: 8.5776 - val_loss: 1.4070\n",
      "Epoch 2/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.5202\n",
      "Epoch 2: val_loss improved from 1.40702 to 1.40133, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 19s 598us/sample - loss: 8.5202 - val_loss: 1.4013\n",
      "Epoch 3/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4981\n",
      "Epoch 3: val_loss improved from 1.40133 to 1.39071, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 19s 601us/sample - loss: 8.4981 - val_loss: 1.3907\n",
      "Epoch 4/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4646\n",
      "Epoch 4: val_loss did not improve from 1.39071\n",
      "31435/31435 [==============================] - 20s 626us/sample - loss: 8.4646 - val_loss: 1.3922\n",
      "Epoch 5/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4003\n",
      "Epoch 5: val_loss improved from 1.39071 to 1.38892, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 19s 613us/sample - loss: 8.4003 - val_loss: 1.3889\n",
      "Epoch 6/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.4000\n",
      "Epoch 6: val_loss did not improve from 1.38892\n",
      "31435/31435 [==============================] - 19s 603us/sample - loss: 8.4000 - val_loss: 1.3892\n",
      "Epoch 7/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3987\n",
      "Epoch 7: val_loss did not improve from 1.38892\n",
      "31435/31435 [==============================] - 21s 677us/sample - loss: 8.3987 - val_loss: 1.3925\n",
      "Epoch 8/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3772\n",
      "Epoch 8: val_loss improved from 1.38892 to 1.38346, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 22s 711us/sample - loss: 8.3772 - val_loss: 1.3835\n",
      "Epoch 9/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3690\n",
      "Epoch 9: val_loss improved from 1.38346 to 1.38129, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 22s 712us/sample - loss: 8.3690 - val_loss: 1.3813\n",
      "Epoch 10/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3584\n",
      "Epoch 10: val_loss did not improve from 1.38129\n",
      "31435/31435 [==============================] - 21s 673us/sample - loss: 8.3584 - val_loss: 1.3857\n",
      "Epoch 11/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3456\n",
      "Epoch 11: val_loss improved from 1.38129 to 1.37998, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 21s 672us/sample - loss: 8.3456 - val_loss: 1.3800\n",
      "Epoch 12/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3315\n",
      "Epoch 12: val_loss did not improve from 1.37998\n",
      "31435/31435 [==============================] - 19s 601us/sample - loss: 8.3315 - val_loss: 1.3835\n",
      "Epoch 13/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3135\n",
      "Epoch 13: val_loss improved from 1.37998 to 1.37943, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 21s 669us/sample - loss: 8.3135 - val_loss: 1.3794\n",
      "Epoch 14/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.3018\n",
      "Epoch 14: val_loss did not improve from 1.37943\n",
      "31435/31435 [==============================] - 22s 707us/sample - loss: 8.3018 - val_loss: 1.3799\n",
      "Epoch 15/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.2959\n",
      "Epoch 15: val_loss did not improve from 1.37943\n",
      "31435/31435 [==============================] - 22s 689us/sample - loss: 8.2959 - val_loss: 1.3817\n",
      "Epoch 16/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.2845\n",
      "Epoch 16: val_loss did not improve from 1.37943\n",
      "31435/31435 [==============================] - 19s 614us/sample - loss: 8.2845 - val_loss: 1.3934\n",
      "Epoch 17/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.2711\n",
      "Epoch 17: val_loss did not improve from 1.37943\n",
      "31435/31435 [==============================] - 19s 617us/sample - loss: 8.2711 - val_loss: 1.3820\n",
      "Epoch 18/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.2705\n",
      "Epoch 18: val_loss did not improve from 1.37943\n",
      "31435/31435 [==============================] - 19s 616us/sample - loss: 8.2705 - val_loss: 1.3872\n",
      "Epoch 19/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.2742\n",
      "Epoch 19: val_loss improved from 1.37943 to 1.37762, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 22s 696us/sample - loss: 8.2742 - val_loss: 1.3776\n",
      "Epoch 20/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 8.2792\n",
      "Epoch 20: val_loss improved from 1.37762 to 1.37618, saving model to ./checkpoints/unknown_person_p22_22.h5\n",
      "31435/31435 [==============================] - 21s 677us/sample - loss: 8.2792 - val_loss: 1.3762\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:38:37.977403: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_36_6/lstm_cell_258/recurrent_kernel/Assign' id:121336 op device:{requested: '', assigned: ''} def:{{{node lstm_36_6/lstm_cell_258/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_36_6/lstm_cell_258/recurrent_kernel, lstm_36_6/lstm_cell_258/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 14:38:43.994406: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_30_6/lstm_cell_252/recurrent_kernel/m/Assign' id:122621 op device:{requested: '', assigned: ''} def:{{{node lstm_30_6/lstm_cell_252/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_30_6/lstm_cell_252/recurrent_kernel/m, lstm_30_6/lstm_cell_252/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31435 samples, validate on 3502 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:38:52.938339: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:39:24.451516: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3611"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:39:46.367046: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.38290, saving model to ./checkpoints/unknown_person_baseline_2_p22_22.h5\n",
      "31435/31435 [==============================] - 35s 1ms/sample - loss: 1.3611 - val_loss: 1.3829\n",
      "Epoch 2/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3514\n",
      "Epoch 2: val_loss did not improve from 1.38290\n",
      "31435/31435 [==============================] - 23s 716us/sample - loss: 1.3514 - val_loss: 1.3830\n",
      "Epoch 3/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3614\n",
      "Epoch 3: val_loss did not improve from 1.38290\n",
      "31435/31435 [==============================] - 21s 660us/sample - loss: 1.3614 - val_loss: 1.3850\n",
      "Epoch 4/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3591\n",
      "Epoch 4: val_loss improved from 1.38290 to 1.38204, saving model to ./checkpoints/unknown_person_baseline_2_p22_22.h5\n",
      "31435/31435 [==============================] - 20s 628us/sample - loss: 1.3591 - val_loss: 1.3820\n",
      "Epoch 5/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3562\n",
      "Epoch 5: val_loss improved from 1.38204 to 1.37917, saving model to ./checkpoints/unknown_person_baseline_2_p22_22.h5\n",
      "31435/31435 [==============================] - 20s 640us/sample - loss: 1.3562 - val_loss: 1.3792\n",
      "Epoch 6/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3519\n",
      "Epoch 6: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 22s 706us/sample - loss: 1.3519 - val_loss: 1.3808\n",
      "Epoch 7/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3515\n",
      "Epoch 7: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 22s 693us/sample - loss: 1.3515 - val_loss: 1.3829\n",
      "Epoch 8/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3503\n",
      "Epoch 8: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 21s 669us/sample - loss: 1.3503 - val_loss: 1.3823\n",
      "Epoch 9/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3492\n",
      "Epoch 9: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 22s 710us/sample - loss: 1.3492 - val_loss: 1.3842\n",
      "Epoch 10/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3483\n",
      "Epoch 10: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 22s 691us/sample - loss: 1.3483 - val_loss: 1.3884\n",
      "Epoch 11/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3480\n",
      "Epoch 11: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 22s 708us/sample - loss: 1.3480 - val_loss: 1.3822\n",
      "Epoch 12/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3482\n",
      "Epoch 12: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 21s 665us/sample - loss: 1.3482 - val_loss: 1.3826\n",
      "Epoch 13/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3489\n",
      "Epoch 13: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 22s 691us/sample - loss: 1.3489 - val_loss: 1.3795\n",
      "Epoch 14/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3455\n",
      "Epoch 14: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 19s 615us/sample - loss: 1.3455 - val_loss: 1.3804\n",
      "Epoch 15/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3454\n",
      "Epoch 15: val_loss did not improve from 1.37917\n",
      "31435/31435 [==============================] - 22s 704us/sample - loss: 1.3454 - val_loss: 1.3805\n",
      "Epoch 16/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3422\n",
      "Epoch 16: val_loss improved from 1.37917 to 1.37820, saving model to ./checkpoints/unknown_person_baseline_2_p22_22.h5\n",
      "31435/31435 [==============================] - 22s 713us/sample - loss: 1.3422 - val_loss: 1.3782\n",
      "Epoch 17/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3427\n",
      "Epoch 17: val_loss improved from 1.37820 to 1.37037, saving model to ./checkpoints/unknown_person_baseline_2_p22_22.h5\n",
      "31435/31435 [==============================] - 21s 671us/sample - loss: 1.3427 - val_loss: 1.3704\n",
      "Epoch 18/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3437\n",
      "Epoch 18: val_loss did not improve from 1.37037\n",
      "31435/31435 [==============================] - 22s 710us/sample - loss: 1.3437 - val_loss: 1.3770\n",
      "Epoch 19/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3411\n",
      "Epoch 19: val_loss did not improve from 1.37037\n",
      "31435/31435 [==============================] - 22s 709us/sample - loss: 1.3411 - val_loss: 1.3763\n",
      "Epoch 20/20\n",
      "31435/31435 [==============================] - ETA: 0s - loss: 1.3416\n",
      "Epoch 20: val_loss did not improve from 1.37037\n",
      "31435/31435 [==============================] - 22s 710us/sample - loss: 1.3416 - val_loss: 1.3816\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [22]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
